NioUIKit/ProfileSettings/Security/Devices/ProfileSettingsSecurityDevicesContainerView.swift

```diff
@@ -0,0 +1,263 @@
+//
+//  ProfileSettingsSecurityDevicesContainerView.swift
+//  Nio
+//
+//  Created by Finn Behrens on 31.03.22.
+//
+
+import MatrixClient
+import MatrixCore
+import NioKit
+import SwiftUI
+
+struct ProfileSettingsSecurityDevicesContainerView: View {
+    @EnvironmentObject var account: MatrixAccount
+    @EnvironmentObject var store: NioAccountStore
+
+    @State private var devices: [MatrixDevice] = []
+    @State private var ownDevice: MatrixDevice?
+
+    @State private var selection = Set<String>()
+    @State private var editMode = EditMode.inactive
+
+    var body: some View {
+        List(selection: $selection) {
+            Section(header: Text("This device")) {
+                if let ownDevice = ownDevice {
+                    NavigationLink {
+                        ProfileSettingsSecurityDeviceDetailView(device: ownDevice, isSelf: true)
+                            .environmentObject(account)
+                    } label: {
+                        ProfileSettingsSecurityDeviceView(device: ownDevice)
+                    }
+                } else {
+                    Text(account.deviceID ?? "Unknown Device")
+                }
+            }
+
+            // TODO: verified/unverified devices sections?
+            Section(header: Text("Devices")) {
+                ForEach(devices, id: \.deviceID) { device in
+                    NavigationLink {
+                        ProfileSettingsSecurityDeviceDetailView(device: device)
+                            .environmentObject(account)
+                    } label: {
+                        ProfileSettingsSecurityDeviceView(device: device)
+                    }
+                    .tag(device.id)
+                }
+                .onDelete(perform: delete)
+            }
+        }
+        .environment(\.editMode, $editMode)
+        .toolbar {
+            ToolbarItem(id: "delete", placement: .bottomBar, showsByDefault: false) {
+                if editMode == .active {
+                    Button("Delete", role: .destructive) {
+                        print("TODO: Mass delete")
+                        withAnimation {
+                            self.editMode = .inactive
+                        }
+                    }
+                    .disabled(self.selection.isEmpty)
+                }
+            }
+        }
+        .refreshable {
+            await self.updateDevices()
+        }
+        .toolbar {
+            ToolbarItem {
+                if editMode == .active {
+                    Button("Cancel") {
+                        self.selection.removeAll()
+                        withAnimation {
+                            self.editMode = .inactive
+                        }
+                    }
+                } else {
+                    Button("Edit") {
+                        withAnimation {
+                            self.editMode = .active
+                        }
+                    }
+                }
```

Comment:
Where possible, it's useful to change parameters on a view rather than switching out the view for a new one:
```suggestion
                Button(editMode.isEditing ? "Cancel" : "Edit") {
                    if editMode.isEditing {
                        self.selection.removeAll()
                        withAnimation {
                            self.editMode = .inactive
                        }
                    } else {
                        withAnimation {
                            self.editMode = .active
                        }
                    }
                }
```